### ISSUE TITLE
data transfer is slow

### ISSUE THREAD

**Issue by jeng1220** [cupy/cupy#1937]
Hi,
I found that transfer data from device to host pinned buffer only achieved 3.2 GB/sec in CuPy.
However, using CUDA C, the same behavior achieved 11.5 GB/sec.

* Conditions
CuPy Version          : 5.1.0
CUDA Build Version    : 10000
CUDA Driver Version   : 10000
CUDA Runtime Version  : 10000
GPU : GTX1080TI
OS : Ubuntu 16.04

* Code to reproduce
CuPy version:
```python
import cupy as cp
import numpy as np
from optparse import OptionParser

pinned_memory_pool = cp.cuda.PinnedMemoryPool()
cp.cuda.set_pinned_memory_allocator(pinned_memory_pool.malloc)
def pin_memory(array):
    mem = cp.cuda.alloc_pinned_memory(array.nbytes)
    ret = np.frombuffer(mem, array.dtype, array.size).reshape(array.shape)
    ret[...] = array
    return ret

def pinned_bandwidth_test(mat_size):
  x_pin = pin_memory(np.empty((mat_size, mat_size), dtype=np.float32))
  x_gpu = cp.empty_like(x_pin)

  stream = cp.cuda.stream.Stream()
  stream.use()

  x_gpu.set(x_pin)
  x_pin = x_gpu.get()

  stream.synchronize()

def main():
  print('use nvprof to watch bandwidth')
  parser = OptionParser()
  parser.add_option('-s', '--size', dest='size',
                    default=4000,
                    help='matrix size')
  (options, args) = parser.parse_args()

  mat_size = int(options.size)
  if mat_size < 1:
    raise Exception('matrix size must be greater than 0')

  pinned_bandwidth_test(mat_size)

if __name__ == '__main__':
  main()
```
CUDA C version:
```c
#include <cuda_runtime.h>
#include <stdlib.h>

void pinned_bandwidth_test()
{
  const int mat_size = 4000;
  size_t mat_size_inbyte = sizeof(float) * mat_size * mat_size;
  float* x_pin;
  float* x_gpu;

  cudaHostAlloc((void**)&x_pin, mat_size_inbyte, cudaHostAllocDefault);
  cudaMalloc((void**)&x_gpu, mat_size_inbyte);

  cudaStream_t stream;
  cudaStreamCreate(&stream);

  cudaMemcpyAsync(x_gpu, x_pin, mat_size_inbyte, cudaMemcpyHostToDevice, stream);
  cudaMemcpyAsync(x_pin, x_gpu, mat_size_inbyte, cudaMemcpyDeviceToHost, stream);
  cudaStreamSynchronize(stream);

  cudaFreeHost(x_pin);
  cudaFree(x_gpu);
  cudaStreamDestroy(stream);
}

int main(int argc, char const *argv[])
{
  pinned_bandwidth_test();
  return 0;
}
```
build CUDA C sample:
```sh
$ nvcc bandwidth_test.c -o bandwidth_test.out
```
* Error messages, stack traces, or logs
nvvp log:
CuPy [link](https://drive.google.com/file/d/1Nvl8yQcs58ZJL4-mWhwKtL13bDvqONkD/view?usp=sharing)
CUDA C [link](https://drive.google.com/file/d/1V7AaKloezd3781pR333zd6HWmppIYXrf/view?usp=sharing)

**Comment by grafi-tt** [comment]
I think this issue can be closed as #1955 is merged.

**Comment by jeng1220** [comment]
Yes, using new usage of `get(out)` API instead of old one can solve this issue. Close issue.
